Code/PgSQL/rdkit/adapter.cpp

```diff
@@ -688,7 +688,14 @@ extern "C" CROMol MolAdjustQueryProperties(CROMol i, const char *params) {
 
 extern "C" char *MolGetSVG(CROMol i, unsigned int w, unsigned int h,
                            const char *legend, const char *params) {
-  RWMol *im = (RWMol *)i;
+  // SVG routines need an RWMol since they change the
+  // molecule as they prepare it for drawing. We don't
+  // want a plain SQL function (mol_to_svg) to have
+  // unexpected side effects, so take a copy and render
+  // (and change) that.
+  const ROMol *input_mol = (ROMol *)i;
+  RWMol input_copy = *input_mol;
```

Comment:
I would prefer using a copy constructor

```
RWMol input_copy(*input_mol);
```

but otherwise this code appears to be fine.

Reply:
@greglandrum Perhaps we should also have a const drawing convenience API?

Reply:
@bp-kelley : you mean something that takes a const mol and calls `prepareMolForDrawing()` with default args?

Reply:
@greglandrum Yes.  I think we hand roll things like that in the wrappers.

Reply:
makes sense. I will put something in this weekend.